components/system_notice/notices.jsx

```diff
@@ -105,6 +107,32 @@ export default [
                 return false;
             }
 
+            return true;
+        },
+    },
+    {
+        name: 'ie11_deprecation',
+        allowForget: false,
+        body: (
+            <FormattedMarkdownMessage
+                id='system_notice.body.ie11_deprecation'
+                defaultMessage='Your browser, IE11, will no longer be supported in an upcoming release. [Find out how to move to another browser in one simple step](!https://forum.mattermost.org/t/mattermost-is-dropping-support-for-internet-explorer-ie11-in-v5-16/7575).'
+            />
+        ),
+        show: (serverVersion) => {
+            const versionNumberSplit = serverVersion.split('.');
+            const versionNumber = [Number(versionNumberSplit[0]), Number(versionNumberSplit[1])];
+
+            // Don't show the notice after v5.16, show a different notice
+            if (versionNumber[0] > 5 || (versionNumber[0] === 5 && versionNumber[1] > 15)) {
+                return false;
+            }
```

Comment:
The above version checking code could be really useful as util. What about generalizing it and adding it here https://github.com/mattermost/mattermost-webapp/blob/master/utils/server_version.jsx ? Then we could write tests against in here https://github.com/mattermost/mattermost-webapp/blob/master/utils/server_version.test.jsx

Reply:
Excellent idea!